src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -1015,7 +1065,10 @@ namespace smt::noodler {
             }
         }
         sat_length_formula = length_formula;
-        add_axiom(length_formula);
+        // It seems thare is problem if the length_formula has quantifiers. In that case we skip adding axioms.
+        if(!expr_cases::has_quantifier(length_formula, m)) {
+            add_axiom(length_formula);
+        }
     }
```

Comment:
If we do not add the axiom, the model might be incorrect, no?

Reply:
Yes, but the solver practically cannot handle quantified formulae. Model generation is not supported for the tag automata procedure, so it is not a big deal now.

Reply:
Ah ok, maybe I would put some WARNING or TODO comment, or some flag that turns off model generation so we get error if we try to get model.

Reply:
Ok, I added a comment.